### PR TITLE
1869 bip32 support

### PR DESCRIPTION
**Pull request by zainen**
add support for use of ledger bip32_ed25519 tz1 addresses 

ability to connect taquito-ledger-signer to ledger using dervationType.bip32_ed25519

### CODE REVIEW

File: integration-tests/ledger-signer.spec.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import TransportNodeHid from "@ledgerhq/hw-transport-node-hid";
 import { ligoSample } from "./data/ligo-simple-contract";
 import { TezosToolkit } from '@taquito/taquito';
 
+// PLEASE NOTE MAY NEED TO TEST ONE TEST AT A TIME
```
Comment (dsawali) [inline integration-tests/ledger-signer.spec.ts:7]: could you elaborate more on this comment?

Comment (zainen) [reply integration-tests/ledger-signer.spec.ts:7]: mainly for the error i would get for 
```
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "PublicKeyRetrievalError: Unable to retrieve Public Key from Ledger".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

Comment (dsawali) [reply integration-tests/ledger-signer.spec.ts:7]: have you tried wrapping the asynchronous parts of your test with a try-catch block? I think that might help with those errors

File: docs/upgrading_guide.md [hunk 2]
```diff
@@ -260,7 +260,7 @@ Ledger support was pre-released in September. Taquito never officially released
 
 This breaking change can impact the users of the pre-released `ledger-signer` package.
 
-We have renamed the enum `DerivationType` members to use the curve name. Now `tz1`, `tz2`, and `tz3` become `ED25519`, `SECP256K1`, and `P256`. This enum is used in the optional `derivationType` parameter of the constructor of the `LedgerSigner` class.
+We have renamed the enum `DerivationType` members to use the curve name. Now `tz1`, `tz2`, and `tz3` become `ED25519`, `BIP32_ED25519`, `SECP256K1`, and `P256`. This enum is used in the optional `derivationType` parameter of the constructor of the `LedgerSigner` class.
```
Comment (roxaneletourneau) [inline docs/upgrading_guide.md:263]: This should not be modified, it was the guide to upgrading to Taquito version 7

File: integration-tests/ledger-signer.spec.ts [hunk 3]
```diff
@@ -185,5 +187,44 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
+
+        const pk = await Tezos.signer.publicKey();
+        const pkh = await Tezos.signer.publicKeyHash();
+
+        expect(pk).toEqual('edpkujVjFVJtb9Z1D7jpSpPMrKzdTRZSRT8E3L26T42vvA6VSv7jND');
+        expect(pkh).toEqual('tz1UpizQ6AGjMeCZCLpuyuL4BSzoUC4XD1QE');
+
+        done();
+      })
+    })
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:228]: Can you add a test where a signature is produced also?

File: packages/taquito-ledger-signer/src/taquito-ledger-signer.ts [hunk 4]
```diff
@@ -81,8 +82,7 @@ export { VERSION } from './version';
  * ```
  * import TransportU2F from "@ledgerhq/hw-transport-u2f";
  * const transport = await TransportU2F.create();
- * const ledgerSigner = new LedgerSigner(transport, "44'/1729'/0'/0'", true, DerivationType.SECP256K1);
- * ```
+ * const ledgerSigner = new LedgerSigner(transport, "44'/1729'/0'/0'", true, DerivationType.SECP256K1);``
```
Comment (roxaneletourneau) [inline packages/taquito-ledger-signer/src/taquito-ledger-signer.ts:85]: Seems like a ` was removed

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: integration-tests/ledger-signer.spec.ts [hunk 5]
```diff
@@ -185,5 +188,88 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
         done();
       });
     })
+
+    describe('Verify that use of a ledger device works with bip32', () => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:192]: Consider replacing `bip32` with `the derivation curve BIP32_ED25519` for clarity

File: integration-tests/ledger-signer.spec.ts [hunk 6]
```diff
@@ -185,5 +188,88 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
+        )
+        const Tezos = new TezosToolkit(rpc);
+        Tezos.setSignerProvider(signer);
+        const pk = await Tezos.signer.publicKey();
+        expect(pk).toEqual('edpkujVjFVJtb9Z1D7jpSpPMrKzdTRZSRT8E3L26T42vvA6VSv7jND');
+        done();
+      })
+    })
+
+    describe('Verify that use of a ledger device works with bip32', () => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:209]: The describe block is the same as the precedent and could be combined into one.

File: integration-tests/ledger-signer.spec.ts [hunk 7]
```diff
@@ -185,5 +188,88 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
+      jest.setTimeout(60000);
+      it('Verify that the pk and pkh is correct', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:211]: This test and the precedent ('Verify that the pk is correct') could be combined into one test.

File: integration-tests/ledger-signer.spec.ts [hunk 8]
```diff
@@ -185,5 +188,88 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
+      it('Verify that the signature is correct', async (done) => {
+        const signer = new LedgerSigner(
+          transport,
+          "44'/1729'/1'/0'",
+          false,
+          DerivationType.BIP32_ED25519
+        )
+        const Tezos = new TezosToolkit(rpc);
+
+        const b = await localForger.forge(rpcToForge)
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:264]: Consider using a variable named more descriptive than `b`

File: integration-tests/data/contract_origination.ts [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+
+export const rpcToForge = { "branch": "BLQ6JyEEzUNkA7WEDSAbyFYLoEK7RVcNcjbjrN5LrmqoUW4qmu3",
+"contents":
+[ { "kind": 'origination',
+   "source": "tz1UpizQ6AGjMeCZCLpuyuL4BSzoUC4XD1QE", "fee": "515",
+   "counter": "340273", "gas_limit": "1533", "storage_limit": "409",
+   "balance": "0",
+   "script":
+     { "code":
```
Comment (roxaneletourneau) [inline integration-tests/data/contract_origination.ts:74]: The code is the same as the one above, can this be combined?

File: integration-tests/package.json [hunk 10]
```diff
@@ -50,7 +52,7 @@
       "/test/"
     ],
     "testPathIgnorePatterns": [
-      "./ledger-signer.spec.ts",
```
Comment (roxaneletourneau) [inline integration-tests/package.json:53]: should not be remove, we don't want those tests to run in the pipeline as it required interaction with a ledger

File: packages/taquito-ledger-signer/test/taquito-ledger-signer.spec.ts [hunk 11]
```diff
@@ -183,4 +183,27 @@ describe('LedgerSigner test', () => {
...
+    const mocksig = Buffer.from(
+      '35c1f3340121965a1350af2082af3c83d4338c23c254591ec7a12fef5d4e9fc2a63f7051508cc41255894fe511cfd11af827e8f8e6c3730c3dd0775aff33dc029000',
+      'hex'
+    );
+    mockTransport.send.mockResolvedValue(mocksig);
+    const signature = await signer.sign('367325bbba406bc3f8c1bf12b27b6e8081064722d3342e34142c172b322ba0426b00c9fc72e8491bd2973e196f04ec6918ad5bcee22d8c0bbcb98d01e85200006760ff228c2c16cbca18bb782a106e51c43a131776f5dfad30ecb5d5e43eccbd6c00c9fc72e8491bd2973e196f04ec6918ad5bcee22dea0abdb98d01c35000a0c21e0000eadc0855adb415fa69a76fc10397dc2fb37039a000');
+    const path = "44'/1729'/1'/0'";
+    const buff = transformPathToBuffer(path);
+    expect(mockTransport.send).toHaveBeenCalledTimes(2);
+    expect(mockTransport.send).toHaveBeenCalledWith(0x80, 0x04, 0x00, 0x03, buff) // TODO ZS check this one
```
Comment (roxaneletourneau) [inline packages/taquito-ledger-signer/test/taquito-ledger-signer.spec.ts:198]: comment `// TODO ZS check this one`?

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: packages/taquito-ledger-signer/test/taquito-ledger-signer.spec.ts (hunk 11 above)
Comment (zainen) [reply packages/taquito-ledger-signer/test/taquito-ledger-signer.spec.ts:198]: right this one was to figure out where these values come form and where to check that these values are correct. 
I adjusted the values to be correct which were similar to the other tests but wanted to make sure that it is correct.

File: integration-tests/ledger-signer.spec.ts [hunk 12]
```diff
@@ -185,5 +189,73 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
+        const pkh = await Tezos.signer.publicKeyHash();
+
+        expect(pk).toEqual('edpkujVjFVJtb9Z1D7jpSpPMrKzdTRZSRT8E3L26T42vvA6VSv7jND');
+        expect(pkh).toEqual('tz1UpizQ6AGjMeCZCLpuyuL4BSzoUC4XD1QE');
+
+        done();
+      })
+    })
+
+    describe('Verify that use of a ledger device works with bip32_ed25519', () => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:215]: empty describe block

Comment (zainen) [reply integration-tests/ledger-signer.spec.ts:215]: thanks!

**Pull request merged**